Integrate gutenberg-mobile release 1.99.0

fluiddot opened the pull request:
## Description
This PR incorporates the 1.99.0 release of gutenberg-mobile.
For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/5944

Release Submission Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

fluiddot committed:
Release script: Update build.gradle gutenbergMobileVersion to ref

fluiddot committed:
Update release notes

fluiddot committed:
Update Gutenberg Mobile ref

geriux commented on RELEASE-NOTES.txt:
```diff
@@ -3,6 +3,8 @@
 -----
 * [*] Fixes the crash that occurs after deleting the site [https://github.com/wordpress-mobile/WordPress-Android/pull/18736]
+* [*] [Jetpack-only] Block editor: Rename "Reusable blocks" to "Synced patterns", aligning with the web editor. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/5885]
+* [**] Block editor: Fix a crash related to Reanimated when closing the editor [https://github.com/wordpress-mobile/gutenberg-mobile/pull/5938]
```
I wonder if we should mark this as internal note 🤔 what do you think?

fluiddot replied:
Yep, good idea. Although it's a crash (i.e. something the user notices), I think the release note doesn't give any value to the user as it's hard to reproduce in the app. I'll update it before merging the PR. Thanks!

geriux replied:
Maybe we could mention something when this is sent to editorial, to highlight that this release includes a fix for random crashes when closing the editor.

fluiddot replied:
Yeah although thinking it twice, maybe we should wait until we get the first crash metrics, once it's released, to confirm that the random crashes are fixed. I'm concerned that the fix doesn't improve those random crashes.

geriux replied:
True, yeah let's do that later on once we confirm it is fixed 👍

geriux reviewed (approved):
LGTM! Friendly reminder to update the Gutenberg mobile commit hash to the release tag. 🚀

fluiddot committed:
Update release notes

Adding `internal` tag to Reanimated crash entry as it's not guaranteed that would fix it.

fluiddot committed:
Update Gutenberg Mobile ref with tag
